In [27]:
import pandas as pd
from sqlalchemy import create_engine

In [28]:
df_csv_src = pd.read_csv('sat-results.csv')
df_csv_src.head()
df_csv_src.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 493 entries, 0 to 492
Data columns (total 11 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   DBN                              493 non-null    object 
 1   SCHOOL NAME                      493 non-null    object 
 2   Num of SAT Test Takers           493 non-null    object 
 3   SAT Critical Reading Avg. Score  493 non-null    object 
 4   SAT Math Avg. Score              493 non-null    object 
 5   SAT Writing Avg. Score           493 non-null    object 
 6   SAT Critical Readng Avg. Score   493 non-null    object 
 7   internal_school_id               493 non-null    int64  
 8   contact_extension                388 non-null    object 
 9   pct_students_tested              376 non-null    object 
 10  academic_tier_rating             402 non-null    float64
dtypes: float64(1), int64(1), object(9)
memory usage: 42.5+ KB


normalize column names (lowercase, replace: ' ', '.')

In [29]:
df_csv_src.columns = df_csv_src.columns.str.lower()
df_csv_src.columns = df_csv_src.columns.str.replace(" ", "_")
df_csv_src.columns = df_csv_src.columns.str.replace(r'[^a-z0-9_]', '', regex=True)
df_csv_src.columns

Index(['dbn', 'school_name', 'num_of_sat_test_takers',
       'sat_critical_reading_avg_score', 'sat_math_avg_score',
       'sat_writing_avg_score', 'sat_critical_readng_avg_score',
       'internal_school_id', 'contact_extension', 'pct_students_tested',
       'academic_tier_rating'],
      dtype='object')

dupe check + dupe deletion

In [30]:
df_csv_src.duplicated().sum()
df_deduped = df_csv_src.drop_duplicates()
df_deduped.count()

dbn                               478
school_name                       478
num_of_sat_test_takers            478
sat_critical_reading_avg_score    478
sat_math_avg_score                478
sat_writing_avg_score             478
sat_critical_readng_avg_score     478
internal_school_id                478
contact_extension                 378
pct_students_tested               363
academic_tier_rating              392
dtype: int64

row_count: 493 > 478
try to automatically convert datatype


In [31]:
df_test = df_deduped.convert_dtypes()
df_test.info()

<class 'pandas.core.frame.DataFrame'>
Index: 478 entries, 0 to 477
Data columns (total 11 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   dbn                             478 non-null    string
 1   school_name                     478 non-null    string
 2   num_of_sat_test_takers          478 non-null    string
 3   sat_critical_reading_avg_score  478 non-null    string
 4   sat_math_avg_score              478 non-null    string
 5   sat_writing_avg_score           478 non-null    string
 6   sat_critical_readng_avg_score   478 non-null    string
 7   internal_school_id              478 non-null    Int64 
 8   contact_extension               378 non-null    string
 9   pct_students_tested             363 non-null    string
 10  academic_tier_rating            392 non-null    Int64 
dtypes: Int64(2), string(9)
memory usage: 45.7 KB


Result: object dtypes got converted to string.

Assessment:
- dbn: okay, since there's alway a char in the middle
- school_name: okay
- num_of_sat_test_takers: num expected
- sat_critical_reading_avg_score: num expected
- sat_math_avg_score: num expected
- sat_writing_avg_score: num expected
- sat_critical_readng_avg_score: same as 'sat_critical_reading_avg_score'? if yes, drop
- internal_school_id: okay
- contact_extension: num expected
- pct_students_tested: num expexted
- academic_tier_rating: okay

Check why sat related columns are interpreted as strings

In [32]:
df_test.groupby('sat_critical_reading_avg_score').count().sort_values(by=['sat_critical_reading_avg_score'], ascending=False)

dbn  school_name  num_of_sat_test_takers  \
sat_critical_reading_avg_score                                             
s                                57           57                      57   
679                               1            1                       1   
636                               1            1                       1   
635                               1            1                       1   
632                               1            1                       1   
...                             ...          ...                     ...   
305                               1            1                       1   
304                               2            2                       2   
300                               1            1                       1   
287                               1            1                       1   
279                               1            1                       1   

                                sat_math_avg_score  sat_writing_avg_score  \
sat_critical_reading_avg_score                                              
s                                               57                     57   
679                                              1                      1   
636                                              1                      1   
635                                              1                      1   
632                                              1                      1   
...                                            ...                    ...   
305                                              1                      1   
304                                              2                      2   
300                                              1                      1   
287                                              1                      1   
279                                              1                      1   

                                sat_critical_readng_avg_score  \
sat_critical_reading_avg_score                                  
s                                                          57   
679                                                         1   
636                                                         1   
635                                                         1   
632                                                         1   
...                                                       ...   
305                                                         1   
304                                                         2   
300                                                         1   
287                                                         1   
279                                                         1   

                                internal_school_id  contact_extension  \
sat_critical_reading_avg_score                                          
s                                               57                 44   
679                                              1                  0   
636                                              1                  1   
635                                              1                  1   
632                                              1                  0   
...                                            ...                ...   
305                                              1                  0   
304                                              2                  2   
300                                              1                  1   
287                                              1                  1   
279                                              1                  1   

                                pct_students_tested  academic_tier_rating  
sat_critical_reading_avg_score                                             
s                                                46                    40  
679              

In [33]:
df_test.query('num_of_sat_test_takers == "s" and sat_critical_reading_avg_score == "s" and sat_math_avg_score == "s" and sat_writing_avg_score == "s"')
#.shape

dbn                                        school_name  \
22          02M392                         MANHATTAN BUSINESS ACADEMY   
23          02M393                          BUSINESS OF SPORTS SCHOOL   
25          02M399         THE HIGH SCHOOL FOR LANGUAGE AND DIPLOMACY   
38          02M427              MANHATTAN ACADEMY FOR ARTS & LANGUAGE   
40          02M437        HUDSON HIGH SCHOOL OF LEARNING TECHNOLOGIES   
41          02M438          INTERNATIONAL HIGH SCHOOL AT UNION SQUARE   
47          02M473                             WASHINGTON IRVING YABC   
77          03M402        THE URBAN ASSEMBLY SCHOOL FOR GREEN CAREERS   
78          03M403                  THE GLOBAL LEARNING COLLABORATIVE   
81          03M417                          FRANK MCCOURT HIGH SCHOOL   
89          03M577  YOUNG ADULT BOROUGH CENTER AT LOUIS D. BRANDEI...   
100         05M362  COLUMBIA SECONDARY SCHOOL FOR MATH, SCIENCE, A...   
111         06M423          HIGH SCHOOL FOR EXCELLENCE AND INNOVATION   
122         07X379                 JILL CHAIFETZ TRANSFER HIGH SCHOOL   
123         07X381                            BRONX HAVEN HIGH SCHOOL   
136         08X269        BRONX STUDIO SCHOOL FOR WRITERS AND ARTISTS   
143         08X367  ARCHIMEDES ACADEMY FOR MATH, SCIENCE AND TECHN...   
144         08X376  ANTONIA PANTOJA PREPARATORY ACADEMY, A COLLEGE...   
148         08X432                          BRONX BRIDGES HIGH SCHOOL   
150         08X507                                     STEVENSON YABC   
153         08X537                            BRONX ARENA HIGH SCHOOL   
167         09X324  BRONX EARLY COLLEGE ACADEMY FOR TEACHING & LEA...   
186         10X319  PROVIDING URBAN LEARNERS SUCCESS IN EDUCATION ...   
201         10X478            LEARNING TO WORK GED AT JOHN F. KENNEDY   
216         11X417   YOUNG ADULT BOROUGH CNTR CHRISTOPHER COLUMBUS HS   
219         11X456        LEARNING TO WORK YABC AT TRUMAN HIGH SCHOOL   
234         12X428            LEARNING TO WORK YABC AT MONROE ACADEMY   
236         12X478                                  THE CINEMA SCHOOL   
237         12X479   BRONX CAREER AND COLLEGE PREPARATORY HIGH SCHOOL   
254         13K527  URBAN ASSEMBLY INSTITUTE OF MATH AND SCIENCE F...   
259         13K616  BROOKLYN HIGH SCHOOL FOR LEADERSHIP AND COMMUN...   
263         14K322                                FOUNDATIONS ACADEMY   
277         14K923                        AUTOMOTIVE HIGH SCHOOL YABC   
286         15K529                WEST BROOKLYN COMMUNITY HIGH SCHOOL   
289         15K667                            SUNSET PARK HIGH SCHOOL   
295         16K688             THE BROOKLYN ACADEMY OF GLOBAL FINANCE   
298         17K467                                       ERASMUS YABC   
315         17K751                         ACADEMY FOR HEALTH CAREERS   
331         19K420                       FRANKLIN K. LANE HIGH SCHOOL   
332         19K431                              THOMAS JEFFERSON YABC   
348         20K609     THE URBAN ASSEMBLY SCHOOL FOR CRIMINAL JUSTICE   
354  21K412/21K411  ABRAHAM LINCOLN YABC/LEARNING TO WORK GED AT A...   
355         21K468                  KINGSBOROUGH EARLY COLLEGE SCHOOL   
380         24Q457  YOUNG ADULT BOROUGH CENTER AT ARTS AND BUSINES...   
388         24Q744                                VOYAGES PREPARATORY   
416         28Q284                         YORK EARLY COLLEGE ACADEMY   
418         28Q325                    HILLSIDE ARTS & LETTERS ACADEMY   
426         28Q686                    QUEENS METROPOLITAN HIGH SCHOOL   
435         29Q326                            CAMBRIA HEIGHTS ACADEMY   
440         30Q286           YOUNG WOMEN'S LEADERSHIP SCHOOL, ASTORIA   
468         75K371                      P.S. 371 s LILLIAN L. RASHKIS   
469         75M035                                           P.S. 035   
470         75Q256                                          P.S. Q256   
472         75R025            SOUTH RICHMOND HIGH SCHOOL I.S./P.S. 25   


57 rows mainly contain 's', possible stands for 'surpressed'. We can safely drop these rows

In [34]:
df_dropped_s = df_test[df_test['num_of_sat_test_takers'] != 's']

In [35]:
df_dropped_s.shape

(421, 11)

check if columns 'sat_critical_reading_avg_score' and 'sat_critical_readng_avg_score', if yes, then drop the column with typo

In [36]:
df_dropped_s.query('sat_critical_reading_avg_score != sat_critical_readng_avg_score')

Empty DataFrame
Columns: [dbn, school_name, num_of_sat_test_takers, sat_critical_reading_avg_score, sat_math_avg_score, sat_writing_avg_score, sat_critical_readng_avg_score, internal_school_id, contact_extension, pct_students_tested, academic_tier_rating]
Index: []

In [37]:
df_cleaned_stage1 = df_dropped_s.drop(columns='sat_critical_readng_avg_score')
df_cleaned_stage1.columns

Index(['dbn', 'school_name', 'num_of_sat_test_takers',
       'sat_critical_reading_avg_score', 'sat_math_avg_score',
       'sat_writing_avg_score', 'internal_school_id', 'contact_extension',
       'pct_students_tested', 'academic_tier_rating'],
      dtype='object')

Now, we have 421 columns remaining
Let Pandas interpret the datatypes again

In [38]:
df_cleaned_stage2 = df_cleaned_stage1.convert_dtypes()
df_cleaned_stage2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 421 entries, 0 to 477
Data columns (total 10 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   dbn                             421 non-null    string
 1   school_name                     421 non-null    string
 2   num_of_sat_test_takers          421 non-null    string
 3   sat_critical_reading_avg_score  421 non-null    string
 4   sat_math_avg_score              421 non-null    string
 5   sat_writing_avg_score           421 non-null    string
 6   internal_school_id              421 non-null    Int64 
 7   contact_extension               334 non-null    string
 8   pct_students_tested             317 non-null    string
 9   academic_tier_rating            352 non-null    Int64 
dtypes: Int64(2), string(8)
memory usage: 37.0 KB


Since, convert_dtypes did not change anything, let's convert them explicitly

In [39]:
numeric_columns = [
    'num_of_sat_test_takers',
    'sat_critical_reading_avg_score', 
    'sat_math_avg_score',
    'sat_writing_avg_score'
]

# Create a copy and convert numeric columns
df_cleaned_stage2[numeric_columns] = df_cleaned_stage2[numeric_columns].apply(pd.to_numeric, errors='coerce')

# Then let pandas optimize the remaining columns
df_cleaned_stage3 = df_cleaned_stage2.convert_dtypes()

# Check the result
df_cleaned_stage3.info()

<class 'pandas.core.frame.DataFrame'>
Index: 421 entries, 0 to 477
Data columns (total 10 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   dbn                             421 non-null    string
 1   school_name                     421 non-null    string
 2   num_of_sat_test_takers          421 non-null    Int64 
 3   sat_critical_reading_avg_score  421 non-null    Int64 
 4   sat_math_avg_score              421 non-null    Int64 
 5   sat_writing_avg_score           421 non-null    Int64 
 6   internal_school_id              421 non-null    Int64 
 7   contact_extension               334 non-null    string
 8   pct_students_tested             317 non-null    string
 9   academic_tier_rating            352 non-null    Int64 
dtypes: Int64(6), string(4)
memory usage: 38.6 KB


In [40]:
df_cleaned_stage3[df_cleaned_stage3['pct_students_tested'].isna()
].head(5)

dbn                              school_name  num_of_sat_test_takers  \
1   01M448      UNIVERSITY NEIGHBORHOOD HIGH SCHOOL                      91   
2   01M450               EAST SIDE COMMUNITY SCHOOL                      70   
5   01M515  LOWER EAST SIDE PREPARATORY HIGH SCHOOL                     112   
8   01M696           BARD HIGH SCHOOL EARLY COLLEGE                     130   
15  02M303               FACING HISTORY SCHOOL, THE                      76   

    sat_critical_reading_avg_score  sat_math_avg_score  sat_writing_avg_score  \
1                              383                 423                    366   
2                              377                 402                    370   
5                              332                 557                    316   
8                              624                 604                    628   
15                             353                 358                    340   

    internal_school_id contact_extension pct_students_tested  \
1               268547              x234                <NA>   
2               236446              x123                <NA>   
5               414951              x345                <NA>   
8               881396              x234                <NA>   
15              612918              <NA>                <NA>   

    academic_tier_rating  
1                      3  
2                      3  
5                      3  
8                   <NA>  
15                  <NA>

Since without the pct_students_tested the whole row does not give us relevant information, we should drop these rows

In [41]:
df_cleaned_stage4 = df_cleaned_stage3[~df_cleaned_stage3['pct_students_tested'].isna()]

In [42]:
df_cleaned_stage4.shape

(317, 10)

Same goes for 'academic_tier_rating'

In [43]:
df_cleaned_stage5 = df_cleaned_stage4[~df_cleaned_stage4['academic_tier_rating'].isna()]

In [44]:
df_cleaned_stage5

dbn                                        school_name  \
0    01M292      HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
3    01M458                          FORSYTH SATELLITE ACADEMY   
4    01M509                            MARTA VALLE HIGH SCHOOL   
6    01M539  NEW EXPLORATIONS INTO SCIENCE, TECHNOLOGY AND ...   
7    01M650                               CASCADES HIGH SCHOOL   
..      ...                                                ...   
464  32K552                          ACADEMY OF URBAN PLANNING   
466  32K556  BUSHWICK LEADERS HIGH SCHOOL FOR ACADEMIC EXCE...   
467  32K564                     BUSHWICK COMMUNITY HIGH SCHOOL   
471  75Q811                                          P.S. Q811   
476  79Q950                                GED PLUS s CITYWIDE   

     num_of_sat_test_takers  sat_critical_reading_avg_score  \
0                        29                             355   
3                         7                             414   
4                        44                             390   
6                       159                             522   
7                        18                             417   
..                      ...                             ...   
464                      67                             342   
466                      23                             347   
467                      24                             359   
471                      32                             429   
476                       8                             496   

     sat_math_avg_score  sat_writing_avg_score  internal_school_id  \
0                   404                    363              218160   
3                   401                    359              427826   
4                   433                    384              672714   
6                   574                    525              697107   
7                   418                    411              297600   
..                  ...                    ...                 ...   
464                 364                    354              695572   
466                 358                    350              658210   
467                 317                    358              898918   
471                 444                    433              467937   
476                 400                    426              544514   

    contact_extension pct_students_tested  academic_tier_rating  
0                x345                 78%                     2  
3                x123                 92%                     4  
4                x123                 92%                     2  
6                <NA>                 78%                     2  
7                <NA>                 92%                     4  
..                ...                 ...                   ...  
464              x234                 92%                     3  
466              <NA>                 92%                     2  
467              <NA>                 92%                     4  
471              x123                 78%                     1  
476              x234                 92%                     2  

[269 rows x 10 columns]

Now, we are at 269 rows, let's identify sat related values, which are out of range

In [45]:
df_cleaned_stage5.query('sat_critical_reading_avg_score < 200 or sat_critical_reading_avg_score > 800 or sat_math_avg_score < 200 or sat_math_avg_score > 800 or sat_writing_avg_score < 200 or sat_writing_avg_score > 800 ')

dbn                                        school_name  \
80   03M415  WADLEIGH SECONDARY SCHOOL FOR THE PERFORMING &...   
422  28Q470                                JAMAICA HIGH SCHOOL   
434  29Q283  PREPARATORY ACADEMY FOR WRITERS: A COLLEGE BOA...   

     num_of_sat_test_takers  sat_critical_reading_avg_score  \
80                       32                             371   
422                      90                             342   
434                      43                             370   

     sat_math_avg_score  sat_writing_avg_score  internal_school_id  \
80                  850                    370              365679   
422                 999                    353              587904   
434                1100                    363              462173   

    contact_extension pct_students_tested  academic_tier_rating  
80               <NA>                 78%                     4  
422              x234                 92%                     3  
434              x345                 85%                     3

Result, only column 'sat_math_avg_score' is affected with 3 rows, let's remove these rows

In [46]:
df_cleaned_stage6 = df_cleaned_stage5.query('not (sat_critical_reading_avg_score < 200 or sat_critical_reading_avg_score > 800 or sat_math_avg_score < 200 or sat_math_avg_score > 800 or sat_writing_avg_score < 200 or sat_writing_avg_score > 800)')

In [47]:
df_cleaned_stage6.info()

<class 'pandas.core.frame.DataFrame'>
Index: 266 entries, 0 to 476
Data columns (total 10 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   dbn                             266 non-null    string
 1   school_name                     266 non-null    string
 2   num_of_sat_test_takers          266 non-null    Int64 
 3   sat_critical_reading_avg_score  266 non-null    Int64 
 4   sat_math_avg_score              266 non-null    Int64 
 5   sat_writing_avg_score           266 non-null    Int64 
 6   internal_school_id              266 non-null    Int64 
 7   contact_extension               214 non-null    string
 8   pct_students_tested             266 non-null    string
 9   academic_tier_rating            266 non-null    Int64 
dtypes: Int64(6), string(4)
memory usage: 24.4 KB



- col 'contact_extension'

Clean column 'pct_students_tested':
- remove '%'
- convert to float and convert to percent

In [48]:
df_cleaned_stage6.loc[:, 'pct_students_tested'] = df_cleaned_stage6['pct_students_tested'].str.replace("%","")

df_cleaned_stage6.loc[:,'pct_students_tested'] = pd.to_numeric(df_cleaned_stage6['pct_students_tested'].astype(str), errors='coerce') / 100


/var/folders/w8/9sd03p8d3msgg0z4xqqw1cbr0000gn/T/ipykernel_60653/2649865181.py:3: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '[0.78 0.92 0.92 0.78 0.92 0.78 0.78 0.92 0.78 0.78 0.78 0.92 0.92 0.85
 0.92 0.92 0.85 0.78 0.92 0.92 0.92 0.85 0.92 0.85 0.85 0.78 0.92 0.92
 0.85 0.78 0.85 0.78 0.78 0.92 0.85 0.85 0.85 0.78 0.85 0.78 0.92 0.92
 0.85 0.78 0.92 0.92 0.92 0.92 0.85 0.92 0.78 0.92 0.92 0.85 0.78 0.92
 0.92 0.78 0.92 0.78 0.78 0.85 0.85 0.92 0.92 0.78 0.78 0.92 0.92 0.78
 0.78 0.78 0.78 0.92 0.85 0.92 0.78 0.85 0.85 0.92 0.78 0.78 0.78 0.78
 0.78 0.92 0.85 0.92 0.78 0.78 0.78 0.85 0.85 0.78 0.85 0.85 0.92 0.85
 0.78 0.85 0.85 0.92 0.85 0.85 0.92 0.78 0.92 0.92 0.85 0.85 0.78 0.78
 0.78 0.78 0.78 0.85 0.92 0.78 0.78 0.78 0.85 0.85 0.92 0.85 0.85 0.85
 0.85 0.85 0.78 0.85 0.92 0.78 0.78 0.78 0.92 0.92 0.92 0.92 0.78 0.78
 0.78 0.78 0.78 0.78 0.85 0.92 0.85 0.78 0.92 0.92 0.85 0.85 0.92 0.85
 0.92 0.78 0.78 0.9

Check if warning had any effects on the column:

In [49]:
df_cleaned_stage6.info()

<class 'pandas.core.frame.DataFrame'>
Index: 266 entries, 0 to 476
Data columns (total 10 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   dbn                             266 non-null    string 
 1   school_name                     266 non-null    string 
 2   num_of_sat_test_takers          266 non-null    Int64  
 3   sat_critical_reading_avg_score  266 non-null    Int64  
 4   sat_math_avg_score              266 non-null    Int64  
 5   sat_writing_avg_score           266 non-null    Int64  
 6   internal_school_id              266 non-null    Int64  
 7   contact_extension               214 non-null    string 
 8   pct_students_tested             266 non-null    float64
 9   academic_tier_rating            266 non-null    Int64  
dtypes: Int64(6), float64(1), string(3)
memory usage: 24.4 KB


Pct_students_tested looks good, finally, let's check column 'contact_extension':

In [50]:
df_cleaned_stage6['contact_extension'].value_counts()

contact_extension
x345    78
x234    70
x123    66
Name: count, dtype: Int64

'contact_extension' contains 3 distinct values and does not add value to the analysis, we'll delete this column

check analytical value of column 'academic_tier_rating'

check analytical value of column 'academic_tier_rating'

In [26]:
df_cleaned_stage6['academic_tier_rating'].value_counts()

academic_tier_rating
4    77
2    68
1    61
3    60
Name: count, dtype: Int64

'academic_tier_rating' contains 4 distinct values, we'll drop this column too

In [56]:
df_final = df_cleaned_stage6.drop(columns=['contact_extension', 'academic_tier_rating'])
df_final.info()

<class 'pandas.core.frame.DataFrame'>
Index: 266 entries, 0 to 476
Data columns (total 8 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   dbn                             266 non-null    string 
 1   school_name                     266 non-null    string 
 2   num_of_sat_test_takers          266 non-null    Int64  
 3   sat_critical_reading_avg_score  266 non-null    Int64  
 4   sat_math_avg_score              266 non-null    Int64  
 5   sat_writing_avg_score           266 non-null    Int64  
 6   internal_school_id              266 non-null    Int64  
 7   pct_students_tested             266 non-null    float64
dtypes: Int64(5), float64(1), string(2)
memory usage: 20.0 KB


### Result: Now we have a dataframe with zero missing values containing 8 columns and 266 rows

Save current dataframe to csv

In [60]:
df_final.to_csv('cleaned_sat_results.csv', index=False)

Create table on postgres db

In [58]:
host = "ep-falling-glitter-a5m0j5gk-pooler.us-east-2.aws.neon.tech"
port = "5432"
database = "neondb"
user = "neondb_owner"
password = "a9Am7Yy5r9_T7h4OF2GN"

# Create a connection engine
engine = create_engine(f"postgresql://{user}:{password}@{host}:{port}/{database}")

In [59]:
df_final.to_sql(
    name='michael_sat_results',       
    con=engine,     
    schema='nyc_schools',
    if_exists='replace',    
    index=False            
)

266